src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -192,23 +192,50 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
 
     private coExpressionCache:CoExpressionCache = new CoExpressionCache(
         q=>({
-            invoke: ()=>{
+            await: () => [this.props.store.entrezGeneIdToReferenceGene],
+            invoke: async ()=>{
                 let threshold = 0.3;
                 if (q.allData) {
                     threshold = 0;
                 }
                 const dataQueryFilter = this.createDataQueryFilterForCoExpression(
                     this.props.store.studyToDataQueryFilter.result![q.profileX.studyId], q.geneticEntityId, q.geneticEntityType);
                 if (dataQueryFilter != undefined) {
-                    // TODO: this sorts by p value asc first, so we can fake
-                    // multi column sort when sorting by q value afterwards. We
-                    // can remove this after implementing multi-sort
-                    return internalClient.fetchCoExpressionsUsingPOST({
+                    let geneset: Geneset | undefined = undefined;
+                    if (q.profileY.molecularAlterationType === AlterationTypeConstants.GENESET_SCORE) {
+                        geneset = await internalClient.getGenesetUsingGET({
+                            genesetId: q.profileY.molecularProfileId
+                        });
+                    }
+
+                    const entrezGeneIdToReferenceGene = this.props.store.entrezGeneIdToReferenceGene.result || {};
+                    const data = await internalClient.fetchCoExpressionsUsingPOST({
                         molecularProfileIdA: q.profileX.molecularProfileId,
                         molecularProfileIdB: q.profileY.molecularProfileId,
                         coExpressionFilter: dataQueryFilter as CoExpressionFilter,
                         threshold
                     });
+
+                    return Promise.resolve(_.reduce(data, (acc, datum) => {
+                        let geneticEntityName: string | undefined = undefined;
+                        if (geneset !== undefined) {
+                            geneticEntityName = geneset.name;
+                        } else if (entrezGeneIdToReferenceGene[datum.geneticEntityId] !== undefined) {
+                            geneticEntityName = entrezGeneIdToReferenceGene[datum.geneticEntityId].hugoGeneSymbol;
+                        }
+
+                        if (geneticEntityName !== undefined) {
```

Comment:
out of curiosity, why do have check for existence of name in order to get cytoband?

Reply:
not required. updated